### PR TITLE
Parametric CalcFactor updates and test

### PR DESCRIPTION
**Pull request by Affie**
towards #382, needs documentation (tracked in JuliaRobotics/IncrementalInference.jl#935)

### CODE REVIEW

File: src/factors/Point2D.jl [hunk 1]
```diff
@@ -18,6 +18,15 @@ function getSample(cfo::CalcFactor{<:PriorPoint2}, N::Int=1)
   return (rand(cfo.factor.Z, N),)
 end
 
+function (s::CalcFactor{<:PriorPoint2})(res::AbstractVector{<:Real}, 	
+                                        meas, 	
+                                        X1)	
+  #	
+  res[1:2] .= meas[1:2] .- X1[1:2]
+
+  nothing
```
Comment (dehann) [inline src/factors/Point2D.jl:27]: we should probably return `res` in these prior cases too.  There is one quirky corner of dispatch for `Mixture` that needs this function.  Likely a better answer possible but this temp fix is okay for now.

File: src/factors/Range2D.jl [hunk 2]
```diff
@@ -23,7 +23,7 @@ function (cfo::CalcFactor{<:Point2Point2Range})(
   XX = lm[1] - (rho[1]*cos(theta[1]) + xi[1])
   YY = lm[2] - (rho[1]*sin(theta[1]) + xi[2])
   res[1] = XX^2 + YY^2
-  res[1]
+  return nothing
```
Comment (dehann) [inline src/factors/Range2D.jl:26]: ah okay, this looks like a step before "return residual to solve dispatch on autodiff".

**Review by dehann — APPROVED** [review]
I think it looks good!

**Comment by dehann** [comment]
xref JuliaRobotics/IncrementalInference.jl#935

File: src/factors/Range2D.jl (hunk 2 above)
Comment (Affie) [reply src/factors/Range2D.jl:26]: That's correct, I got it running, see updated IIF#467

**Comment by Affie** [comment]
Thanks for linking 935, I'll track docs there and then this can close #382

**Pull request merged**